coredocument/coredocument.go

```diff
@@ -149,6 +160,44 @@ func GetDocumentSigningTree(document *coredocumentpb.CoreDocument) (tree *proofs
 	return tree, nil
 }
 
+// GetDocumentSigningTree returns the merkle tree for the signing root
+func GetDocumentSigningTree(document *coredocumentpb.CoreDocument) (tree *proofs.DocumentTree, err error) {
+	h := sha256.New()
+
+	// coredoc tree
+	coreDocTree, err := GetDocumentCDTree(document)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the signing tree with data root and coredoc root as siblings
+	t2 := proofs.NewDocumentTree(proofs.TreeOptions{EnableHashSorting: true, Hash: h})
+	tree = &t2
+	err = tree.AddLeaves([]proofs.LeafNode{
```

Comment:
@lucasvo any objection to this instead of hashing them together here it self? I feel this is cleaner for proofs, since tree structure is clear.